### PR TITLE
Soft-delete ScratchOrgs and track CCI logs on them

### PR DESCRIPTION
**Pull request by wlonk**
Based off of `delete-project`, because soft-delete logic was already implemented there.

fox

### CODE REVIEW

**Comment by wlonk** [comment]
@davisagli this is provisional, and should probably wait for #231, simply because it is based off the soft-delete logic there. But take a look and let me know if it addresses your needs.

**Comment by wlonk** [comment]
[Soft-delete ScratchOrgs and store logfile from cci.](https://trello.com/c/UmrmIRii/176-soft-delete-scratchorgs-and-store-logfile-from-cci)

**Comment by wlonk** [comment]
[Soft-delete ScratchOrgs and store logfile from cci.](https://trello.com/c/UmrmIRii/176-soft-delete-scratchorgs-and-store-logfile-from-cci)

File: metecho/api/sf_run_flow.py [hunk 1]
```diff
@@ -325,6 +325,7 @@ def run_flow(*, cci, org_config, flow_name, project_path, user):
         traceback = p.stdout.decode("utf-8")
         logger.warning(traceback)
         raise Exception(_last_line(traceback) or _last_line(err))
+    return out
```
Comment (davisagli) [inline metecho/api/sf_run_flow.py:328]: @wlonk Notice that this code is detecting a non-zero returncode and raising an exception on line 327 (the exception message ends up getting displayed on the frontend). I want to be sure the log is stored in that case too -- in fact that's the case when we are likely to go looking for the log.

And then we need to talk about the source for the log. It should be read from the log file I mentioned in the trello card, not from the cci process's stdout. The log file includes a copy of both stdout and stderr, as well as some additional context such as what command was run.

**Pull request merged**